### ISSUE TITLE
Formatting `Amount` with particular fractional digit separators

### ISSUE THREAD

**Issue by RCasatta** [rust-bitcoin/rust-bitcoin#1478]
I still have to decide if I love it or hate it, but I think implementing it in the `Amount` type makes sense and lets wallet implementors and users decide if they like it.

![image](https://user-images.githubusercontent.com/6470319/208050132-14c65bc3-a6f6-46e9-b399-7ec050a13a37.png)

At rust-bitcoin level I think we have to provide the option to use configurable separators.

More info on the format:
https://bitcoin.design/guide/designing-products/units-and-symbols/
https://medium.com/@mark.nugent.iv/grouping-bitcoins-fractional-digits-an-idea-whose-time-has-come-22d9dad8ac51

**Comment by sanket1729** [comment]
> lets wallet implementors and users decide if they like it.

concept ACK. As long it is well-specified and easy to maintain. Hopefully, we don't have new standards coming often.

**Comment by Kixunil** [comment]
I find this format confusing because of the use of both `.` and `,`.

>  I think we have to provide the option to use configurable separators.

This would be a hard requirement for me to ACK this.

That being said I do like the decimal part.

**Comment by Kixunil** [comment]
I think this has two parts:

## Make separators configurable

Add these methods on `amount::Display`

* `fn decimal_separator(separator: char) -> Self`
* `fn thousands_separator(separator: Option<char>) -> Self`, `None` means not separated; default: `None`
* `fn fraction_group_separator(separator: Option<char>) -> Self`, `None` means not separated; default: `None`

Have a **documentd** `debug_assert` in `fmt::Display` impl ensuring `thousands_separator != Some(decimal_separator) && fraction_group_separators != Some(decimal_separator)`

## Support satcomma

Have `display_satcomma(self)` method on `{Signed}Amount` and explain it implies `Bitcoin` unit and sets thousands and fraction group separators to space by default. This implies the part above needs to be done first.

Also explain somewhere we don't display currency symbol - the caller has to handle this.

**Comment by apoelstra** [comment]
Agreed that both separators need to be configurable. Super weird that this image shows a confusing mix of commas, periods and spaces, when none of the supporting blog posts (or the text on the rest of the page) do this. I think the point is to illustrate that you're supposed to make these configurable for localization purposes :).

As far as code strategy, I like Kix's plan.

**Comment by dpc** [comment]
I've seen many amount formatting ideas, but this one is the most confusing one yet, by a large margin. :D
